### ISSUE TITLE
Buffer overflow in iOS - calc

### ISSUE THREAD

**Issue by bino38** [CollaboraOnline/online#7708]
When I try to edit a „larger“ file (An Ods file created on a Mac with Libre Office (7.5). The file was often edited with Collabora app. The Ods file is about 800 kb in size, with 160 columns and 40 rows, half with formulas to calculate the data. The file also contains charts.) in iOS with calc, the Collabora app crashes every time since version 23.05.5.
Before 23.05.5, I could edit the same file on iOS.
When I try to edit this file, the app freezes. First all columns and rows (with all entries) turn white and can neither be displayed nor entered or edited. After that, I have to close the app and restart it.

I use an iPad Pro (11 inch, 4. Generation) with iOS 17.1.1.

**Comment by mmeeks** [comment]
@Minion3665 any thoughts ? and any chance you can mail the file to Skyler ?

**Comment by Tex2002ans** [comment]
This was originally from the Collabora Forum:

- https://forum.collaboraonline.com/t/ios-buffer-overflow-in-calc-diagrams-not-usable/2193

(There may be more info in that thread too.)

- - -

@bino38 And yes, like mmeeks asked, it would be very helpful if you could:

- Share the problematic ODS file.

That would help the team figure out what's freezing up the iOS version (and get it fixed). :)

**Comment by bino38** [comment]
Can you explain to me how I can upload the file here?

**Comment by Minion3665** [comment]
> Can you explain to me how I can upload the file here?

@bino38 sure! There should be a button above the comment box that looks like a paperclip 
![paperclip button](https://github.com/CollaboraOnline/online/assets/34243578/ee9755d3-c11f-40b9-b02c-cddf749e4aa4)
If you click it, you'll be able to insert a file

**Comment by bino38** [comment]
Oh, I have to use the website. The icon does not exist in the GitHub app.
Thanks

This is one of these problematic files:
[Grundsicherung für Arbeitssuchende.ods](https://github.com/CollaboraOnline/online/files/15246684/Grundsicherung.fur.Arbeitssuchende.ods)

**Comment by Minion3665** [comment]
> Oh, I have to use the website. The icon does not exist in the GitHub app. Thanks
> 
> This is one of these problematic files: [Grundsicherung für Arbeitssuchende.ods](https://github.com/CollaboraOnline/online/files/15246684/Grundsicherung.fur.Arbeitssuchende.ods)

Thanks very much for providing the file, I've been taking a look at it and I'm seeing the broken behavior too!

From a little profiling it seems that the file is taking a really long time to render - perhaps the content that hasn't rendered is what you're seeing when everything turns white...

[nohang.pdf](https://github.com/CollaboraOnline/online/files/15247457/nohang.pdf)
[hang.pdf](https://github.com/CollaboraOnline/online/files/15247458/hang.pdf)

**Comment by mmeeks** [comment]
Interesting profile; it would be interesting @Minion3665 to put some logging output into ScOutput::DrawStrings - and also ScOutputData::DrawEditStandard - to dump the content of the strings we're printing; I wonder if there is some cockup whereby we're rendering things multiple times. The sheet doesn't look at the upper end of hyper-complexity, so ... would be worth digging into that I think =)

**Comment by smehrbrodt** [comment]
Is this still reproducible?

**Comment by rpear** [comment]
Yes this still reproducible, it appears to be the same as Minion3665 says on May 8, 2024
I tested the spreadsheet on an iPhone 12 Pro Max, iOS 26.6.1 and an iPad 9th gen, iPadOS 26.6.1, Version: 26.04.4.0 build 4. It takes over 2 minutes to load from local storage, I was then able to edit a few cells without anything obviously odd occurring.  When loading it takes a few seconds to render on the screen - as if it has loaded, the screen is showing the far-right columns, but other cells to the left are blank, these take 2 minutes to appear.
